sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/GatewayAddressCache.java

```diff
@@ -84,14 +85,18 @@
     private volatile Pair<PartitionKeyRangeIdentity, AddressInformation[]> masterPartitionAddressCache;
     private volatile Instant suboptimalMasterPartitionTimestamp;
 
+    private final ConcurrentHashMap<URI, Set<PartitionKeyRangeIdentity>> serverPartitionAddressToPkRangeIdMap;
```

Comment:
in my understanding in .Net version we are doing the cache update in the global cache (all regions).
But here we are doing it per region (per GatewayAddressCache).

am I right?

Reply:
In .net version, the cache is also per region (per GatewayAddressCache). 

The difference is around when exception happens, which cache to update:
In .net version, all caches (all regions) will be checked and updated.
In current java, we will get the endpoint from `URI serviceEndpoint = this.endpointManager.resolveServiceEndpoint(request);`, so we only update the cache for that specific region.